aten/src/ATen/native/cuda/Shape.cu

```diff
@@ -257,7 +257,15 @@ void parallel_cat(Tensor &out, const TensorList &inputs, int64_t dimension,
 } // namespace
 
 Tensor cat_cuda(TensorList inputs, int64_t dimension) {
-  Tensor out = at::empty({0}, inputs.front().options());
+  // Ideally we would like to obtain the type using `result_type`
+  // But some linker error prevent us from using it here
+  // ScalarType high_type = at::native::result_type(inputs);
```

Comment:
hmmm... that makes it inconsistent with the CPU impl though. if it won't link this way, another possibility might be a function that computes the common dtype and re-dispatches to a function that accepts a scalar type as a parameter.

Reply:
Sorry, I am not understanding your proposal.

Reply:
I was able to fix the linker error thanks to @take-cheeze